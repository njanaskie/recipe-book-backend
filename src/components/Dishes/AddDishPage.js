import React, { useReducer } from 'react'
import DishForm from './DishForm'
import IngredientsContext from '../../../context/ingredients-context'
import ingredientsReducer from '../../reducers/ingredients'

const AddDishPage = () => {
    const [ingredients, dispatch] = useReducer(ingredientsReducer, [])

    return (
        <IngredientsContext.Provider value={{ ingredients, dispatch }}>
            <h1>Add Dish</h1>
            <DishForm />
        </IngredientsContext.Provider>
    )
}

export default AddDishPage