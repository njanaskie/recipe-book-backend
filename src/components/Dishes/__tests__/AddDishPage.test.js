import React from 'react'
import { shallow } from 'enzyme'
import AddDishPage from '../AddDishPage'

test('should render AddDishPage correctly',() => {
    const wrapper = shallow(<AddDishPage />)
    expect(wrapper).toMatchSnapshot()
})