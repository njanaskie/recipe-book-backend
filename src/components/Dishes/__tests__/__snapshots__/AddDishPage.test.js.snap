// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddDishPage correctly 1`] = `
<ContextProvider
  value={
    Object {
      "dishDispatch": [Function],
      "dishes": Array [],
    }
  }
>
  <ContextProvider
    value={
      Object {
        "dispatch": [Function],
        "ingredients": Array [],
      }
    }
  >
    <h1>
      Add Dish
    </h1>
    <AddDishHome />
  </ContextProvider>
</ContextProvider>
`;
